src/textual/reactive.py

```diff
@@ -156,21 +190,24 @@ def __set_name__(self, owner: Type[MessageTarget], name: str) -> None:
 
     def __get__(self, obj: Reactable, obj_type: type[object]) -> ReactiveType:
         _rich_traceback_omit = True
-        value: _NotSet | ReactiveType = getattr(obj, self.internal_name, _NOT_SET)
-        if isinstance(value, _NotSet):
-            # No value present, we need to set the default
-            init_name = f"_default_{self.name}"
-            default = getattr(obj, init_name)
-            default_value = default() if callable(default) else default
-            # Set and return the value
-            setattr(obj, self.internal_name, default_value)
-            if self._init:
-                self._check_watchers(obj, self.name, default_value)
-            return default_value
+
+        self._initialize_reactive(obj, self.name)
+
+        value: ReactiveType
+        compute_method = getattr(self, f"compute_{self.name}", None)
+        if compute_method is not None:
+            old_value = getattr(obj, self.internal_name)
+            value = getattr(obj, f"compute_{self.name}")()
+            setattr(obj, self.internal_name, value)
+            self._check_watchers(obj, self.name, old_value)
+        else:
+            value = getattr(obj, self.internal_name)
         return value
 
     def __set__(self, obj: Reactable, value: ReactiveType) -> None:
         _rich_traceback_omit = True
+
+        self._initialize_reactive(obj, self.name)
         name = self.name
         current_value = getattr(obj, name)
```

Comment:
I could be wrong, and this is confusing, but:

1. Shouldn't this be `self.internal_name` rather than `name`?
2. Specifically, using `name` -- doesn't that invoke the compute method again, because you're accessing through the `__get__` in the descriptor? Isn't that against the contract? (compute is invoked only when value changes)